doc/src/atom_style.rst

```diff
@@ -10,7 +10,7 @@ Syntax
 
    atom_style style args
 
-* style = *angle* or *atomic* or *body* or *bond* or *charge* or *dipole* or  *dpd* or *edpd* or *electron* or *ellipsoid* or *full* or *line* or *mdpd* or *molecular* or *oxdna* or *peri* or *smd* or *sph* or *sphere* or *spin* or *tdpd* or *tri* or *template* or *hybrid*
+* style = *angle* or *atomic* or *body* or *bond* or *charge* or *dipole* or  *dpd* or *edpd* or *electron* or *ellipsoid* or *full* or *line* or *mdpd* or *molecular* or *oxdna* or *peri* or *smd* or *sph* or *sphere* or *sphere/bpm* or *spin* or *tdpd* or *tri* or *template* or *hybrid*
```

Comment:
Why atom style `sphere/bpm`? Why not just atom style `bpm`?
We don't have any other pair styles with a "slash" (except for suffixed versions for kokkos) and there are others that are supersets (e.g. atomic -> bond -> angle -> molecular -> full, or atomic -> charge -> full)

Reply:
There are many BPMs in the literature and they combine many different types of bond and atom styles including point particles, finite particles, aspherical particles, etc. Since there isn't such a thing as a universal bpm atom style, I avoided using that label. I tried to make the name derivative of atom style sphere since particles are rotational spheres with a fixed density/diameter. Is there an appropriate way to use a two word name? 

I'll also try and think of one word names.

Reply:
> Is there an appropriate way to use a two word name?

The "natural" approach to that from within LAMMPS would be to use something like  `atom_style hybrid sphere bpm` and then have the bpm atom style only implement what is missing from sphere.  After the most recent refactoring of the AtomVec classes by @sjplimp the difference in the realization of, say, `atom_style full` versus `atom_style hybrid charge molecular` has become very small.

Mind you I have not really looked at the code but just at the documentation, so I cannot make any statements about possible complications in the implementation.

Mind you, my time is currently rather limited, so I won't be able to do a proper code review and my own tests for a while.
There are a few windows of opportunity. But then again, I am currently waiting for the PR to clear the integration and unit testing before even trying to figure out what needs looking at. Since this touches a lot of core functionality, it will be required to look very carefully to minimize the risk that existing functionality is broken.

Reply:
I see, in this case it'd be `atom_style hybrid sphere bond X` where `X` would be a new atom style that just stores what is missing (namely a quaternion). I'll think about possible alternate solutions.

I understand completely about time. I know this PR has some hairy changes - like the reworking of special bond/history bits in neighbor lists.

I fixed an overlooked merge conflict I introduced earlier today and that resolved the build issue. I can also look at the issues with the unit tests but I admittedly do not know where to find details on what failed.